internal/shellgen/tmpl/glibc-patch.nix.tmpl

```diff
@@ -0,0 +1,64 @@
+{
+  description = "Patches packages to use a newer version of glibc";
+
+  inputs = {
+    nixpkgs-glibc.url = "{{ .NixpkgsGlibcFlakeRef }}";
+
+    {{- range $name, $flakeref := .Inputs }}
+    {{ $name }}.url = "{{ $flakeref }}";
+    {{- end }}
+  };
+
+  outputs = args@{ self, nixpkgs-glibc {{- range $name, $_ := .Inputs -}}, {{ $name }} {{- end }} }:
+    {{ with .Outputs -}}
+    let
+      # Initialize each nixpkgs input into a new attribute set with the
+      # schema "pkgs.<input>.<system>.<package>".
+      #
+      # Example: pkgs.nixpkgs-80c24e.x86_64-linux.python37
+      pkgs = builtins.mapAttrs (name: flake:
```

Comment:
where is this `name` used? i.e. in the Example provided above, where is the `nixpkgs-80c24e` being set?

Reply:
It's not used anywhere, but it's a required argument to `mapAttrs`.

This is just mapping the existing names from the flake inputs (set by the template) to a new attribute set. The new set has the configured nixpkgs. So `nixpkgs-80c24e → pkgs.nixpkgs-80c24e` where `pkgs.nixpkgs-80c24e = (import nixpkgs-80c24e { ... })`. 

This way we don't have to try and guess if an input resolves to nixpkgs in Go. Instead we can check at eval-time with Nix.